### DISCUSSION TITLE
[BP#12] Limiter-le-recours-aux-plugins

### DISCUSSION THREAD

**Discussion by DocRoms** [cnumr/best-practices#28] [Bonnes pratiques] [answered]
La règle [Limiter le recours aux plugins](https://github.com/cnumr/best-practices/blob/main/chapters/bonne_pratique_012_fr.md#limiter-le-recours-aux-plugins) stipule ceci : 

---
**Règle**
Eviter d’utiliser des plug-ins (Flash Player, machines virtuelles Java et Silverlight, etc.), car certains consomment beaucoup de ressources (cycles CPU et mémoire vive). C’est notamment le cas du Flash Player d’Adobe, à tel point qu’Apple a décidé de ne pas installer cette technologie sur ses terminaux mobiles afin de garantir une meilleure autonomie. 
Préférer les technologies standards comme HTML5, ECMAScript, etc.

**Exemple**
Pour réaliser un back office basé sur du drag and drop, préférer l’utilisation de jQuery à celle de Flash.

---

Je trouve que cette règle possède trois soucis : 

1️⃣ Flash est déprécié, son support s'est arrêté en Décembre 2020.

2️⃣ Dans la phrase 
```Préférer les technologies standards comme HTML5, ECMAScript, etc.```
 on parle d'ECMAScript qui est certe, une technologie qui forme la base de JavaScript, mais beaucoup de personnes ne le savent pas et confondent ça avec les normes ECMA ( ex: ECMA-262) . Nous devrions peut-être plutôt dire `Préférer les technologies standards comme HTML5, ECMAScript, JAVAScript, etc.` ou simplement remplacé `ECMAScript` par `JavaScript` pour éviter la confusion.

3️⃣ L'exemple n'est pas (ou n'est plus) cohérent:  
```préférer l’utilisation de jQuery à celle de Flash.```
Flash est déprécié, JQuery est une libraire qu'il n'est pas utile d'installer pour faire tourner un site internet. Dans l'idéal il ne faut utiliser **_ni l'un ni l'autre_**. 
Il faudrait peut-être avoir un exemple d'actualité comme
```Préférer une utilisation des standards du Web (HTML/CSS/Javascript) qu'un plugin```

**Comment by arnolem** [comment]
Je pense qu'il faut complétement supprimer cette règle pour de nombreuses raisons :
 - Aujourd'hui, il est impossible de faire un site web avec Flash, Silverlight ou NPAPI-JAVA (Ni Chrome, ni Firefox, ni Opera, ni Edge ne le supporte).
 - Ce type de contenu reste ultra-marginal sur le web et ne représente pas une piste sérieuse de réduction de la consommation
 - Si ce type de contenu existe encore en interne dans quelques sociétés, leur 1er problème n'est pas l'éco-conception.

**Comment by tbroyer** [comment]
+1 pour la suppression

Par contre @DocRoms tu peux préciser quelle différence tu fais entre ECMAScript et JavaScript ?

**Reply by DocRoms** [reply]
La suppression me va 😅 .

JavaScript est un language informatique implémentant des règles ECMA dédiée au scripting ;) 

Il y a pas mal de discussion sur le sujet, et c'est assez confus pour une bonne partie du monde :) C'est pour cela que je proposait de ne parler que de JS, qui est le language utilisé implémentant les règles ES.

ECMA de défini lui même comme : 
`Ecma International est une association industrielle dédiée à la normalisation des systèmes d'information et de communication`

Sources : 
 - https://www.ecma-international.org/ 
 - https://qastack.fr/programming/912479/what-is-the-difference-between-javascript-and-ecmascript
 - https://fr.wikipedia.org/wiki/ECMAScript

**Comment by didierfred** [comment]
+1 pour la suppression

**Comment by ACTLEM** [comment]
D'avis pour supprimer cette BP. A voir pour ajouter une sur "Respecter les standards du web" comme tu le proposes avec des exemples courants de non standardisation, mais j'avoue que je n'ai pas d'exemple récent qui me sont venus en tête.

**Comment by tbroyer** [comment]
Est-ce qu'on ne devrait pas remplacer cette BP obsolète par une autre d'actualité mais liée aux mêmes problématiques: l'utilisation d'un canvas pour "dessiner" l'interface utilisateur (cf. Flutter for Web; exemples: https://code.irobot.com/ et https://demo.invoiceninja.com/ qui utilisent le CanvasKit renderer, ou https://vaccine.enroll.health.gov.za/ qui utilise le HTML renderer). CanvasKit utilise canvas et WASM, qui sont pourtant des "standards du web".

Je ne comprend pas l'usage de Flutter for Web, vs. faire une PWA qui aura les mêmes cibles (on pourra même la wrapper dans une enveloppe native pour la publier sur les stores d'application qui le requièrent si c'est si important)

**Comment by Christophe-Cl** [comment]
Au delà du descriptif de la règle qui doit être en effet mise à jour, ce qui me gène est déjà le titre avec le terme "plugin" : est-ce une extension de CMS ? de navigateur web ? de logiciel ? d'outil framework de développement ?

Suggestion : orienter le sujet sur les librairies-tiers, donc les composants externes, qui ajoutent du js/css/code et sur lesquelles ni les concepteurs/développeurs ni les utilisateurs n'ont de maitrise environnementale ou des données collectées.

**Comment by Fred-artwai** [comment]
+1 pour la suppression

**Comment by ACTLEM** [comment, marked as answer]
Suppression de la BP: Issue https://github.com/cnumr/best-practices/issues/245
